### PR TITLE
[CDF-842] & [CDF-843] & [CDF-845]

### PR DESCRIPTION
**Pull request by diogofscmariano**
[CDF-842] - Added type and protected to the viewParameters object

@afrjorge please view

### CODE REVIEW

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.js [hunk 1]
```diff
@@ -270,19 +284,21 @@ define([
     },
 
     /**
-     * The dashboard's debug level.
+     * @summary The dashboard's debug level.
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.js]: remove 's -> "The dashboard debug level."

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.js [hunk 2]
```diff
@@ -163,19 +167,25 @@ define([
...
+     *
+     * @protected
+     * @type {cdf.dashboard.RefreshEngine}
+     */
+    refreshEngine: undefined,
+
+
+    /**
+     * @summary Determines if components and params are retrieved from the _window_ object or from the dashboard instance
+     * @description Determines if components and params are retrieved from the _window_ object or from the dashboard instance
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.js]: replace "params" with "parameters", " _window_" with " `window` " and add punctuation.

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.js [hunk 3]
```diff
@@ -35,7 +35,11 @@ define([
      * @extends {@link http://dean.edwards.name/weblog/2006/03/base/|Base}
      * @extends {@link http://backbonejs.org/#Events|Backbone.Events}
      * @classdesc Base abstract dashboard class.
-     * @param {Object} options Object that can contain the context, storage and view.
+     * @param {Object} [options] Object that can contain the context, storage and view.
+     * @param {Object} [options.context] The context of the dashboard retrieved from the server.
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.js]: add links? {@link cdf.dashboard.Dashboard#context|context} (same below if you agree)

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.js [hunk 4]
```diff
@@ -304,18 +320,19 @@ define([
     },
 
     /**
-     * Sets the globalContext value.
+     * @summary Sets the globalContext value.
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.js]: add links {@link cdf.dashboard.Dashboard#globalContext|globalContext}

File: core-js/src/main/javascript/cdf/dashboard/Dashboard.js [hunk 5]
```diff
@@ -304,18 +320,19 @@ define([
...
-     * @ignore
      */
     setGlobalContext: function(globalContext) {
       this.globalContext = globalContext;
     },
 
     /**
-     * Gets the current webapp path.
+     * @summary Gets the current webapp path.
+     * @description Gets the current webapp path.
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/dashboard/Dashboard.js]: add "webAppPath: undefined," and doclet somewhere :)

**Comment by diogofscmariano** [comment]
@afrjorge pr updated, please review

**Comment by diogofscmariano** [comment]
[CDF-843] - Added documentation for Dashboard methods and properties
[CDF-842] - Added type and protected to the viewParameters object
[CDF-845] - Added summaries for UnmanagedComponent

@afrjorge please review

File: core-js/src/main/javascript/cdf/components/UnmanagedComponent.js [hunk 6]
```diff
@@ -554,7 +571,8 @@ define([
     // ------------
 
     /**
-     * Creates and sets the component's current query given its definition, and, optionally, query options.
+     * @summart Creates and sets the component's current query given its definition, and, optionally, query options.
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/components/UnmanagedComponent.js]: summary typo

**Pull request merged**